### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by coderacheal**
### **In this pull request**

Create a method called current_user
Create forms to perform the following functions:
Creates a Post on behalf of the current_user
Create a comment on behalf of the current_user
Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -9,4 +11,31 @@ def show
...
+    @post.author = current_user
+
+    if @post.save
+      # if post saved, redirect to the user's posts and render success message
+      redirect_to user_posts_path(current_user), notice: 'Post created successfully.'
+    else
+      # if post not saved, render new template and render error message
+      render :new, notice: 'Post could not be created.'
+    end
+  end
```
Comment (muneebulrehman) [inline app/controllers/posts_controller.rb]: For every post that I create, on the **User Index**,  **User Show**, and the **Post Index** pages, it is displaying double the actual amount of posts for a user. Eg., here I have created two posts, but the Number of posts for the user is showing 4. Kindly fix this issue. ⬇️
image

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -2,17 +2,22 @@
...
 
   <div class="com_on_Post">
-  <h3>comments:</h3>
+  <h3>comments</h3>
     <ul>
       <% @post_comments.each do |comment| %>
         <li><%= comment.author.name %>: <%= comment.text %></li>
       <% end %>
     </ul>
+    <%= link_to "Add comment", new_user_post_comment_path(user_id: @current_user.id, post_id: @post.id) %>
```
Comment (muneebulrehman) [inline app/views/posts/show.html.erb:21]: On the **Post Index** page, it should only display the recent 5 comments on the post. You shoud be using `recent_comments` method in the **Post** model to do this job. However, in your case, it is displaying all the comments of a post. Kindly fix this issue. ⬇️
image

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @coderacheal , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Great job adding the forms
- Great job adding the necessary routes.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @coderacheal , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Implemented change requested to ensure the app displays the correct number of posts
✔️ Implemented the change requested to ensure the app displays only five comments in the post index page

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**